### PR TITLE
[TwigComponent] Improve ComponentFactory performances

### PR DESCRIPTION
**Pull request by smnandre**
ComponentFactory Quick Optimization

Minor change with.. mid impact
(on large number of renders)

### CODE REVIEW

**Comment by smnandre** [comment]
false positive for fabbot

**Comment by norkunas** [comment]
When running let's say frankenphp, this will always stay in memory?

File: src/TwigComponent/src/ComponentFactory.php [hunk 1]
```diff
@@ -141,37 +145,40 @@ public function get(string $name): object
...
-        }
-
         if ($component instanceof AnonymousComponent) {
             $component->mount($data);
 
             return;
         }
 
-        $parameters = [];
+        if (!\array_key_exists($component::class, self::$mountMethods)) {
```
Comment (Kocal) [inline src/TwigComponent/src/ComponentFactory.php]:
```suggestion
        if (null === (self::$mountMethods[$component::class] ?? null)) {
```

**Review by Kocal — APPROVED** [review]

**Comment by Kocal** [comment]
Thanks for this optimization!

However I'm thinking, not now but maybe for a next PR, but do you think things can be more improved if this "find mounts methods" process was done during a cache warmer and persisted in cache? 
I don't speak about storing `ReflectionMethod` in cache (I don't think it is possible), but storing that if `$component::class` **has** `mount` method or not: 
```diff
    private function mount(object $component, array &$data): void
    {
        if ($component instanceof AnonymousComponent) {
            $component->mount($data);

            return;
        }
 
+       if (!self::$hasMountMethod[$component::class]) {
+           return;
+       }

        if (!\array_key_exists($component::class, self::$mountMethods)) {
            try {
                $mountMethod = self::$mountMethods[$component::class] = (new \ReflectionClass($component))->getMethod('mount');
            } catch (\ReflectionException) {
                self::$mountMethods[$component::class] = false;

                return;
            }
        }
```
WDYT?

**Comment by smnandre** [comment]
> However I'm thinking, not now but maybe for a next PR [...]  WDYT?

Absolutely! 

I'm moving step by step, a bit in the blind. The biggest change is yet to come (this week hopefully). This is why i'm only doing work on private/internal methods for now.

Once we have the results i'm targetting, a clean will be very welcome (required?)..  We will then see how to name these classes / methods, and how we can rearrange the whole thing

👍

**Comment by smnandre** [comment]
> When running let's say frankenphp, this will always stay in memory?

Would it be a problem ? Genuine question here :)

**Comment by norkunas** [comment]
Depends on how much memory it will use. ResetInterface could be used to cleanup after request

**Comment by smnandre** [comment]
Updated!

**Review by Kocal — APPROVED** [review]

**Pull request merged**